octobot_evaluators/channels/evaluators.py

```diff
@@ -0,0 +1,135 @@
+# cython: language_level=3
+#  Drakkar-Software OctoBot-Evaluators
+#  Copyright (c) Drakkar-Software, All rights reserved.
+#
+#  This library is free software; you can redistribute it and/or
+#  modify it under the terms of the GNU Lesser General Public
+#  License as published by the Free Software Foundation; either
+#  version 3.0 of the License, or (at your option) any later version.
+#
+#  This library is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+#  Lesser General Public License for more details.
+#
+#  You should have received a copy of the GNU Lesser General Public
+#  License along with this library.
+
+from octobot_channels.channels.channel import Channel
+from octobot_channels.constants import CHANNEL_WILDCARD
+from octobot_channels.consumer import Consumer
+from octobot_channels.producer import Producer
+from octobot_commons.logging.logging_util import get_logger
+
+
+class EvaluatorsChannelConsumer(Consumer):
+    pass
+
+
+class EvaluatorsChannelProducer(Producer):
+    # noinspection PyMethodOverriding
+    async def send(self,
+                   matrix_id,
+                   data,
+                   evaluator_name=CHANNEL_WILDCARD,
+                   evaluator_type=CHANNEL_WILDCARD,
+                   exchange_name=CHANNEL_WILDCARD,
+                   cryptocurrency=CHANNEL_WILDCARD,
+                   symbol=CHANNEL_WILDCARD,
+                   time_frame=CHANNEL_WILDCARD):
+        for consumer in self.channel.get_filtered_consumers(matrix_id=matrix_id,
+                                                            data=data,
+                                                            evaluator_name=evaluator_name,
+                                                            evaluator_type=evaluator_type,
+                                                            exchange_name=exchange_name,
+                                                            cryptocurrency=cryptocurrency,
+                                                            symbol=symbol,
+                                                            time_frame=time_frame):
+            await consumer.queue.put({
+                "matrix_id": matrix_id,
+                "evaluator_name": evaluator_name,
+                "evaluator_type": evaluator_type,
+                "exchange_name": exchange_name,
+                "cryptocurrency": cryptocurrency,
+                "symbol": symbol,
+                "time_frame": time_frame,
+                "data": data
+            })
+
+
+class EvaluatorsChannel(Channel):
```

Comment:
We should add an api to stop this channel (for backtesting end)

Reply:
Is implemented in https://github.com/Drakkar-Software/OctoBot-Evaluators/pull/52. Can't be done here.